odoo/service/security.py

```diff
@@ -12,9 +12,12 @@ def compute_session_token(session, env):
     self = env['res.users'].browse(session.uid)
     return self._compute_session_token(session.sid)
 
-def check_session(session, env):
+
+def check_session(session, env, request=None):
```

Comment:
Hello, couldn't this method be made to only receive `request` as other params can be read directly from it?
It seems to never be called without a `request`.

Reply:
It is called without it, in one place, in the bus or something if I remember correctly. I had the same comment and then saw that extreme case in one place where it was called without it and without possibility to call it with it.

Reply:
In `_dispatch_bus_notifications`, called in `get_messages`, called in `_serve_forever` where a `WebsocketRequest` is created on the next line.
I wonder if we could move the creation of the request one line up to avoid this limitation.
cc @tsm-odoo

Reply:
I didn't want to take the risk now for this task, as it was a bit out of scope of this, but do it in a next iteration.

Reply:
Then you do not do that `_update_device` for websocket calls.
We analyzed the stack of all possible ways where a session can be used, and this `check_session` is the common place for all ways: HTTP calls, websocket calls, XMLRPC calls.

Reply:
ah yes, websocket, indeed